diff --git src/CMakeLists.txt src/CMakeLists.txt
index d3e105f48..a71011f1d 100644
--- src/CMakeLists.txt
+++ src/CMakeLists.txt
@@ -7,8 +7,6 @@ if (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-unused-private-field -Wno-unknown-warning-option")
 endif()
 
-option(BUILD_WERROR "Build with -Werror" ON)
-
 if(BUILD_WERROR)
   if (LINUX)
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror")
